### PR TITLE
Expose remote peer's static key

### PR DESCRIPTION
**Pull request by str4d**
Based on #8.

### CODE REVIEW

File: src/session.rs [hunk 1]
```diff
@@ -127,6 +128,14 @@ impl Session {
         }
     }
 
+    /// Get the remote static key that was possibly encrypted in the first payload
+    pub fn get_remote_static(&self) -> Option<&[u8; MAXDHLEN]> {
```
Comment (mcginty) [inline src/session.rs]: on second peek, returning an array reference sized at MAXDHLEN is a misleading public-facing API. Might be more clean to return a `Option<&[u8]>` that's properly sliced to the length based on the chosen DH function?

Comment (str4d) [reply src/session.rs]: Mmm, probably. I encountered a similar issue with one of the Noise extensions I'm working on, where I end up with `[u8; MAXHASHLEN]` in the public-facing API. I haven't yet decided how to solve that one, but this API has an unambiguous meaning relative to the spec, so it should be safe to give users a trimmed slice. I'll make that change.

File: src/handshakestate.rs [hunk 2]
```diff
@@ -284,13 +284,14 @@ impl HandshakeState {
         Ok(payload_len)
     }
 
-    pub fn get_remote_static(&self) -> Option<&[u8; MAXDHLEN]> {
-        self.rs.as_option_ref()
+    pub fn get_remote_static(&self) -> Option<&[u8]> {
+        self.rs.as_option_ref().map(|rs| &rs[..self.dh_len()])
```
Comment (str4d) [inline src/handshakestate.rs]: I modified this function to return a slice, as it wasn't being used anywhere else, and it meant we didn't need to expose `dh_len` inside `Session`.

**Comment by mcginty** [comment]
Looks great. Squashed your two commits, rebased, and merged. Thanks @str4d!

**Pull request closed without merging**